source/docs/non_linear_arithmetic/non_linear_basic.rs

```diff
@@ -0,0 +1,136 @@
+// testcases and examples are ported from below two files 
```

Comment:
What's the purpose of this file? Is there a specific reason it isn't part of the tests?

Reply:
I moved to docs because this lemma `LemmaMulUpperBound` failed on my machine, but it seems to work in the CI machine. (I will move to `examples`.)

Since `integer_ring` is available with `int`, users need to check bounds when they want to prove properties about bounded integers. That file is to show an example of that usage.